crates/storage/provider/src/post_state/mod.rs

```diff
@@ -193,6 +193,26 @@ impl PostState {
     /// TODO: This function hides an expensive operation (bloom). We should probably make it more
     /// explicit.
     pub fn receipts_root(&self, block: BlockNumber) -> H256 {
+        #[cfg(feature = "optimism")]
+        {
+            // There is a minor bug in op-geth and op-erigon where in the Regolith hardfork,
+            // the receipt root calculation does not include the deposit nonce in the receipt
+            // encoding. This will be fixd in the next hardfork, however for now, we must strip
+            // the deposit nonce from the receipts before calculating the receipt root.
```

Comment:
```suggestion
            // There is a minor bug in op-geth and op-erigon where in the Regolith hardfork,
            // the receipts root calculation does not include the deposit nonce in the receipt
            // encoding. This will be fixed in the next hardfork, however, for now, we must strip
            // the deposit nonce from the receipts before calculating the receipts root.
```